### ISSUE TITLE
Losing connection

### ISSUE THREAD

**Issue by exeljb** [thejeffreystone/hassio_addons#12]
Does anyone have any ideas why I've been experiencing disconnection issues.  I am not thinking it's a hardware issue as it's always been resolved by either restarting home assistant (not the machine, just the home assistant supervisor) or the Acurite to Home Assistant add-on. 

I'm using a virtual machine to run home assistant so there might be some quirks with how virtual box handles usb connections.  I also have a wyze sense hub connected to usb and haven't had connection issues with that.  

It seems like the acurite to home assistant add-on just quits and doesn't attempt to search for a reconnect opportunity. 
<img width="1536" alt="Screen Shot 2021-06-11 at 8 35 23 PM" src="https://user-images.githubusercontent.com/59851334/121762819-ee640100-cafd-11eb-9b57-6f4875be3351.png">

**Comment by thejeffreystone** [comment]
hmm. I am not sure what is going on there. It does look like its loosing connection to the usb dongle. I'll have to see if there is a way to get it to attempt to reconnect instead of just dying. But I suspect that is the RTL_433 that is dying. I might have to wrap it in something so that it can monitor the state.

**Comment by staniel1881** [comment]
Hi, 

I had this -5 error also and it buzzled me, even killed one of my RTLs after two months completely. Finally I added an 0.5m usb -extension so the RTL is not directly connected into the PC/pi USB-Port and the error was gone. Without it, I had the docker crash 6-7 times a day. I even set up notification with node-red for 15min timeouts to remind me to manually restart docker. 

I especially got the error when I physically touched something on my setup.  I was planning also to restart the docker every time it crashed with node-red, however the error disappeared with adding the extension and therefore didn't implement it.  If you get this error constantly prepare for backup RTL_433 dongle, you might just like me end up at one point without an working RTL, at the beginning restart works, but later you need to replug the dongle, until its finally not recognizable any more.

**Comment by exeljb** [comment]
That was a great tip. So about 10 days ago, I unplugged the dongle from the computers USB port and plugged it into a hub (since I didn’t readily have an extension available). While unplugging it from the computer, I found the dongle was surprisingly hot to the touch. Now that it’s on the hub, it’s only a bit warmer than ambient, so maybe it was on the verge of roasting itself.  During these 10 days I haven’t had the issue I brought up above and performance has been great!
